src/test/java/io/mosip/mimoto/service/CredentialServiceTest.java

```diff
@@ -0,0 +1,135 @@
+package io.mosip.mimoto.service;
+
+import com.google.gson.Gson;
+import io.mosip.mimoto.dto.IssuersDTO;
+import io.mosip.mimoto.dto.LogoDTO;
+import io.mosip.mimoto.dto.mimoto.*;
+import io.mosip.mimoto.exception.ApiNotAccessibleException;
+import io.mosip.mimoto.service.impl.CredentialServiceImpl;
+import io.mosip.mimoto.service.impl.IssuersServiceImpl;
+import io.mosip.mimoto.util.RestApiClient;
+import io.mosip.mimoto.util.Utilities;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.Velocity;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.mosip.mimoto.service.IssuersServiceTest.getIssuerDTO;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest
+public class CredentialServiceTest {
+
+    @Mock
+    IssuersService issuersService = new IssuersServiceImpl();
+    @InjectMocks
+    CredentialServiceImpl credentialService = new CredentialServiceImpl();
+
+    @Mock
+    Utilities utilities;
+
+    @Mock
+    public RestApiClient restApiClient;
+
+    List<String> issuerConfigRelatedFields = List.of("additional_headers", "authorization_endpoint","authorization_audience", "token_endpoint", "proxy_token_endpoint", "credential_endpoint", "credential_audience", "redirect_uri");
+
+    @Before
+    public void setUp() throws Exception {
+        IssuersDTO issuers = new IssuersDTO();
+        issuers.setIssuers(List.of(getIssuerDTO("Issuer1", Collections.emptyList()), getIssuerDTO("Issuer2", Collections.emptyList())));
+        Mockito.when(issuersService.getAllIssuersWithAllFields()).thenReturn(issuers);
+    }
+    static CredentialsSupportedResponse getCredentialSupportedResponse(String credentialSupportedName){
```

Comment:
Move this response into json file.
Please refer develop-java21 branch. We have added response as json file in test resources and then can use JSONAssert.

Reply:
The Assertion we are currently doing will do compare all the values at once, and you dont have to put assertion on each values of the POJO. extracted the utility method to separate file and keep it so test file will be clean with just test.